src/iter/test.rs

```diff
@@ -654,7 +654,7 @@ fn check_partial_cmp_late_nan_direct() {
 }
 
 #[test]
-fn check_partial_cmp_late_nane_to_seq() {
+fn check_partial_cmp_late_none_to_seq() {
```

Comment:
The correction should be related to NaN:
```suggestion
fn check_partial_cmp_late_nan_to_seq() {
```